modules/folder/main.tf

```
@@ -48,3 +62,30 @@ resource "google_compute_firewall_policy_association" "default" {
   name              = var.firewall_policy.name
   firewall_policy   = var.firewall_policy.policy
 }
+
+resource "google_assured_workloads_workload" "folder" {
+  count                     = var.assured_workload_config == null ? 0 : 1
+  compliance_regime         = var.assured_workload_config.compliance_regime
+  display_name              = var.assured_workload_config.display_name
+  location                  = var.assured_workload_config.location
+  organization              = var.assured_workload_config.organization
+  enable_sovereign_controls = try(var.assured_workload_config.enable_sovereign_controls, null)
+  labels                    = try(var.assured_workload_config.labels, {})
+  partner                   = try(var.assured_workload_config.partner, null)
+  dynamic "partner_permissions" {
+    for_each = try(var.assured_workload_config.partner_permissions, null) == null ? [] : [""]
+    content {
+      assured_workloads_monitoring = try(var.assured_workload_config.partner_permissions.assured_workloads_monitoring, null)
+      data_logs_viewer             = try(var.assured_workload_config.partner_permissions.data_logs_viewer, null)
+      service_access_approver      = try(var.assured_workload_config.partner_permissions.service_access_approver, null)
```

```suggestion
      assured_workloads_monitoring = var.assured_workload_config.partner_permissions.assured_workloads_monitoring
      data_logs_viewer             = var.assured_workload_config.partner_permissions.data_logs_viewer
      service_access_approver      = var.assured_workload_config.partner_permissions.service_access_approver
```

I'd skip `try`s here, as you already proven that `var.assured_workload_config.partner_permissions` is not null in `for_each`